core/src/main/java/org/mule/runtime/core/policy/SuccessSourcePolicyResult.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core.policy;
+
+import org.mule.runtime.core.api.Event;
+
+import java.util.Map;
+
+/**
+ * Result of a successful execution of a {@link SourcePolicy}.
+ *
+ * It contains the {@link Event} result of the flow execution and the response parameters and error response parameters to be sent
+ * by the source.
+ *
+ * @since 4.0
+ */
+public class SuccessSourcePolicyResult {
+
+  private final Event flowExecutionResult;
+  private final Map<String, Object> responseParameters;
+  private final MessageSourceResponseParametersProcessor messageSourceResponseParametersProcessor;
+
+  /**
+   * Creates a new successful result from a policy execution.
+   *
+   * @param flowExecutionResult the result of the flow execution.
+   * @param responseParameters the response parameters to be sent by the source.
+   */
+  public SuccessSourcePolicyResult(Event flowExecutionResult, Map<String, Object> responseParameters,
+                                   MessageSourceResponseParametersProcessor messageSourceResponseParametersProcessor) {
+    this.flowExecutionResult = flowExecutionResult;
+    this.responseParameters = responseParameters;
+    this.messageSourceResponseParametersProcessor = messageSourceResponseParametersProcessor;
+  }
+
+  /**
+   * @return the result of the flow execution.
+   */
+  public Event getFlowExecutionResult() {
+    return flowExecutionResult;
+  }
+
+  /**
+   * @return the response parameters to be sent by the source.
+   */
+  public Map<String, Object> getResponseParameters() {
+    return responseParameters;
+  }
+
+  /**
+   * @param event the event to use as source for generating the error response parameters.
+   * @return the set of parameters to execute the function that sends the failure response.
+   */
+  public Map<String, Object> createErrorResponseParameters(Event event) {
```

Comment:
Because even though this is a successful response it may fail when evaluating the response parameters. If that's the case the exception strategy needs to get executed and the error response parameters must be executed with the result of the exceptions strategy.